bob-backend/src/pearl/disk_controller.rs

```diff
@@ -465,15 +464,11 @@ impl DiskController {
 
     pub(crate) async fn get_alien(&self, op: Operation, key: BobKey) -> Result<BobData, Error> {
         if *self.state.read().await == GroupsState::Ready {
-            let vdisk_group = self.find_group(&op).await;
-            if let Ok(group) = vdisk_group {
-                group.get(key).await
+            let mut results = self.get_alien_results_unordered(&op, |g| async move { g.get(key).await }).await.unwrap_or(vec![]);
+            results.sort_by_key(|d| d.meta().timestamp());
```

Comment:
Pointless vector allocation and sorting. Function `get_alien_results` can return the latest result

Reply:
Actually, `get_alien_results_unordered` is too generic. Just read the list of groups and run `for` loop over that list. Separate function make the code harder to read

Reply:
This function is reused in exist and delete where we need all results

Reply:
Both look like `fold` though, so I'll add it as argument to avoid allocations in other calls